packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/utils.tsx

```diff
@@ -74,7 +74,36 @@ export const isServerSpan = spanHasTag.bind(null, 'span.kind', 'server');
 
 const isErrorBool = spanHasTag.bind(null, 'error', true);
 const isErrorStr = spanHasTag.bind(null, 'error', 'true');
-export const isErrorSpan = (span: Span) => isErrorBool(span) || isErrorStr(span);
+
+export interface IErrorSpanInfo {
+  isError: boolean; // true if this span OR its descendants have errors
+  selfError: boolean; // true if only this span has errors
+}
+
+export const isErrorSpan = (span: Span): IErrorSpanInfo => {
```

Comment:
why do we need to change the signature of this function? It asks explicitly about a single span, so the only possible answer is yes or not, there's no need to split the answer into two bools.